qiskit_nature/algorithms/pes_samplers/bopes_sampler.py

```diff
@@ -132,6 +173,15 @@ def sample(self, problem: BaseProblem, points: List[float]) -> BOPESSamplerResul
         if self._driver.molecule is None:
             raise QiskitNatureError("Driver MUST be configured with a Molecule.")
 
+        # force the initialization of the solver before starting the computation
+        if isinstance(self._gss, MinimumEigensolverFactory):
+            if self._qubit_converter is None:
+                raise QiskitNatureError(
+                    "When using MinimumEigensolverFactory " "you must specify a qubit converter"
+                )
+            prepare_problem(problem, self._qubit_converter)
```

Comment:
`prepare_problem` returns the mapped `main_operator` but you're not using that here.

EDIT: I see that this is not necessary but maybe add a comment why.

Reply:
In this case we don't need the main_operator. Prepare_problem is called because we need a problem with second quantisation applied in order to create the solver. The main_operator is used later, when the real calculations are performed. 

I must admit that this is not as clean as it should be and I did 3 implementations to get to this solution. The problem is that MinimumEigensolverFactory produces the solver based on a problem with second quantisation applied and GroundStateSolver is a wrapper which does the second quantisation and calculates the ground state on a raw problem(in the same method). Bopes sampler adds some parameters to the solver GroundStateSolver has, but in case of MinimulEigensolverFactory the creation of the solver must be delayed... So, the row above force the second quantisation on the problem in order to create a solver to gather the parameters that bopes sampler calculates.

One cleaner solution would have been to create a GroundStateSolver instance based on the MinimumEigensolverFactory and keep all the code. This solution didn't work as it introduces cyclic dependencies.

Reply:
I actually meant that you add a short comment in the code such that, when somebody else looks at the code in the far future, the comment helps to understand why it is neccessary.